scripts/apps/authoring/attachments/actions.ts

```diff
@@ -0,0 +1,89 @@
+import {gettext} from 'core/utils';
+
+export const INIT_ATTACHMENTS = 'INIT_ATTACHMENTS';
+export function initAttachments(item) {
+    return (dispatch, geState, {deployConfig, attachments}) => {
+        const payload = {
+            maxSize: deployConfig.getSync('attachments_max_size'),
+            maxFiles: deployConfig.getSync('attachments_max_files'),
+            files: [],
+        };
+
+        attachments.byItem(item)
+            .then((files) => {
+                payload.files = files;
+            })
+            .finally(() => {
+                dispatch({type: INIT_ATTACHMENTS, payload: payload});
+            });
+    };
+}
+
+export const ADD_ATTACHMENTS = 'ADD_ATTACHMENTS';
+export function selectFiles(files) {
+    return (dispatch, getState, {$scope, notify, superdesk}) => {
+        const state = getState();
+
+        if (Array.isArray(files) && files.length > 0 && !state.editor.isLocked) {
+            if (files.length + state.attachments.files.length > state.attachments.maxFiles) {
+                notify.error(gettext('Sorry, too many files selected.'));
```

Comment:
Ioan already merged translating plurals, so you can make the error messages more informative.

```
gettextCatalog.getPlural(
    state.attachments.maxFiles,
    "Too many files selected. Only one file is allowed.",
    "Too many files selected. Only {{$count}} files are allowed."
)
```